Cache/Main.lean

```diff
@@ -82,9 +88,38 @@ def main (args : List String) : IO Unit := do
   let hashMap := hashMemo.hashMap
   let goodCurl ← pure !curlArgs.contains (args.headD "") <||> validateCurl
   if leanTarArgs.contains (args.headD "") then validateLeanTar
+
+  let noteManyStaleFiles := do
+    let numberAllCacheFiles := (← getLocalCacheSet).size
+    let currentHits := hashMap.size
+    -- If at least 80% of all current cache files are orphaned, warn the user.
+    -- (IOW, the total cache size is at least 5x the number of current hits.)
+    if 5 * currentHits < numberAllCacheFiles then
+      IO.println "note: currently, more than 80% of all files in the cache are not used \
+      (they might be useful for other versions of mathlib, or orphaned): \
+      to increase disk space, run cache clean to delete these files"
```

Comment:
If I have 6 disjoint caches of Mathlib, all of which I'm actively using in different project/branches, I will get this message? That doesn't seem useful, since that is almost always the situation I'm in.

Reply:
Where's the racecar emoji...

Reply:
That question (about the cache *location* for different projects) is a good one: I'm not sure what the answer is.
@arthurpaulino Do you know more/can you help here, please?

Reply:
I was going to suggest also looking at file atime to see whether large portions of the cache were stale but I know atime behavior can be unreliable depending on the filesystem so I'm not sure how well it'd work in practice, but might be worth checking.